src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,50 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exeption.InvalidUserDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_FIELD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws InvalidUserDataException {
+        if (userValidator(user) && foundUserInDatabase(user)) {
+            storageDao.add(user);
+        }
+        return user;
+    }
+
+    @Override
+    public boolean foundUserInDatabase(User user) {
+        if (storageDao.get(user.getLogin()) == null) {
+            return false;
+        }
+        return storageDao.get(user.getLogin()).getLogin().equals(user.getLogin());
+    }
+
+    @Override
+    public boolean userValidator(User user) throws InvalidUserDataException {
```

Comment:
bad name of method -> it should describe action
```suggestion
    public boolean userValidator(User user) throws InvalidUserDataException {
```